hack/pp-cm-helper.sh

```diff
@@ -0,0 +1,235 @@
+    2. Explicitly defined enviroment vars
+    3. Fixed/Fetched/Existing values
+EOF
+}
+
+error_exit() {
+    echo "Error: $1" >&2
+    exit 1
+}
+
+# Loop through all the arguments
+while getopts ":yv:c:h" opt; do
+    case ${opt} in
+        c ) [[ ${OPTARG} =~ ^(sev|tdx)$ ]] && DISABLECVM=false && CTYPE="${OPTARG}" || error_exit "unknown trusted platform type" ;;
+        h ) usage && exit 0;;
+        v ) [[ "${OPTARG}" == *"="* ]] && export ${OPTARG} && custom_vars+=(${OPTARG%%=*});;
+        y ) export YES=true;;
+	\? ) echo "Invalid option: -$OPTARG" >&2 && usage && exit 1;;
+    esac
+done
+
+# Expected Configuration Keys
+common_vars=("CLOUD_PROVIDER" "VXLAN_PORT" "PROXY_TIMEOUT" "DISABLECVM")
+
+aws_vars=("PODVM_INSTANCE_TYPE" "PODVM_INSTANCE_TYPES" "AWS_REGION" "AWS_SUBNET_ID" "AWS_VPC_ID" "AWS_SG_IDS")
+aws_optional=("PODVM_AMI_ID")
+
+azure_vars=("AZURE_INSTANCE_SIZE" "AZURE_INSTANCE_SIZES" "AZURE_SUBNET_ID" "AZURE_NSG_ID" "AZURE_REGION" "AZURE_RESOURCE_GROUP")
+azure_optional=("AZURE_IMAGE_ID")
+
+
+#### Functions
+
+function exportVars() {
+    for pair in "$@"; do
+        key=${pair%%=*}
+        [[ -z "${!key}" ]] && export "$pair" # && echo "Defined: $pair" || echo "Skip: $key"
+    done
+}
+
+function verifyAndSetVars() {
+   arr=("$@")
+   for i in "${arr[@]}"; do
+       local varName="$i"
+       local varValue=${!i}
+       local userInput
+
+       # skip if verfied already
+       [[ -f  "$ENV_FILE" ]] && grep -q "^$varName=" "$ENV_FILE" && continue
+       if [[ -n $YES ]]; then
+           echo "${varName} [${varValue}]: ${varValue}"
+       else
+           read -p "${varName} [${varValue}]: " userInput
+       fi
+       if [[ "${userInput}" != "drop" ]]; then
+           echo "$varName=${userInput:-$varValue}" >> ${ENV_FILE}
+       else
+           echo "dropping ${varName}"
+       fi
+   done
+
+}
+
+function getIMDSDefaults() {
+    local name=imds-defaulter
+    ${CLI} apply -f - <<EOF
+apiVersion: batch/v1
+kind: Job
+metadata:
+  name: ${name}
+  namespace: openshift-sandboxed-containers-operator
+spec:
+  ttlSecondsAfterFinished: 30
+    #completions: 1 # same as default
+    #parallelism: 1 # same as default
+  template:
+    spec:
+      hostNetwork: true
+      restartPolicy: Never
+      containers:
+      - image: 'registry.redhat.io/openshift4/ose-cli'
+        name: ${name}
+        command:
+        - 'bash'
+        - '-c'
+        - |
+          [[ -n \$(oc get cm peer-pods-cm -n openshift-sandboxed-containers-operator 2>/dev/null) ]] && echo "ConfigMap already exist, skipping..." && exit 0
+          provider=\$(oc get infrastructure -n cluster -o jsonpath='{.items[*].status.platformStatus.type}' | awk '{print tolower(\$0)}' | tr -d '"') && echo "cloud provider: \${provider}"
+          if [ \${provider} == "aws" ]; then
+              export MAC=\$(curl -m 30 -s --show-error "http://169.254.169.254/latest/meta-data/mac")
+              cat <<EOS >> /tmp/cm.env
```

Comment:
```suggestion
              cat <<EOS >> ${ENV_FILE}
```

and other users of `/tmp/cm.env` as well.

Reply:
all of these within the job yaml are actually running out of shell (in a pod)